kyo-prelude/shared/src/main/scala/kyo/Emit.scala

```diff
@@ -21,24 +21,25 @@ object Emit:
         given CanEqual[Ack, Ack] = CanEqual.derived
         inline given Flat[Ack]   = Flat.unsafe.bypass
 
+        def apply(maxItems: Int): Ack =
+            if maxItems <= 0 then Stop else Continue(maxItems)
+
         extension (ack: Ack)
             def maxItems(n: Int): Ack =
-                ack match
-                    case Stop         => Stop
-                    case Continue(n0) => Math.min(n0, Math.max(0, n))
+                Ack(Math.min(ack, n))
 
         /** Indicates to continue emitting values */
         opaque type Continue <: Ack = Int
         object Continue:
             def apply(): Continue              = Int.MaxValue
-            def apply(maxItems: Int): Continue = Math.max(0, maxItems)
+            def apply(maxItems: Int): Continue = Math.max(1, maxItems)
```

Comment:
I think a compile-time check wouldn't help much since the number is typically in a scope where its value is dynamic. I like the idea of having only `Ack.apply`. If the provided value is `<= 0`, then it's implicitly a stop.